lms/static/js/spec/student_profile/learner_profile_fields_spec.js

```diff
@@ -0,0 +1,276 @@
+            });
+
+            it("can't upload/remove image if parental consent required", function() {
+                var imageView = createImageView(true, false, 64, 16, '');
+                imageView.render();
+
+                spyOn(imageView, 'clickedUploadButton');
+                spyOn(imageView, 'clickedRemoveButton');
+
+                expect(imageView.$('.u-field-upload-button').css('display') === 'none').toBeTruthy();
+                expect(imageView.$('.u-field-remove-button').css('display') === 'none').toBeTruthy();
+
+                imageView.$('.u-field-upload-button').click();
+                imageView.$('.u-field-remove-button').click();
+
+                expect(imageView.clickedUploadButton).not.toHaveBeenCalled();
+                expect(imageView.clickedRemoveButton).not.toHaveBeenCalled();
+            });
+
+            it("can't upload image on others profile", function() {
+                var imageView = createImageView(false);
+                imageView.render();
+
+                spyOn(imageView, 'clickedUploadButton');
+                spyOn(imageView, 'clickedRemoveButton');
+
+                expect(imageView.$('.u-field-upload-button').css('display') === 'none').toBeTruthy();
+                expect(imageView.$('.u-field-remove-button').css('display') === 'none').toBeTruthy();
+
+                imageView.$('.u-field-upload-button').click();
+                imageView.$('.u-field-remove-button').click();
+
+                expect(imageView.clickedUploadButton).not.toHaveBeenCalled();
+                expect(imageView.clickedRemoveButton).not.toHaveBeenCalled();
+            });
+
+            it("shows message if we try to navigate away during image upload/remove", function() {
+                var imageView = createImageView(true, false);
+                spyOn(imageView, 'onBeforeUnload');
+                imageView.render();
+
+                // Initialize jquery file uploader
+                imageView.$('.upload-button-input').fileupload({
+                    url: Helpers.IMAGE_UPLOAD_API_URL,
+                    type: 'POST',
+                    add: imageView.fileSelected,
+                    done: imageView.imageChangeSucceeded,
+                    fail: imageView.imageChangeFailed
+                });
+
+                // Add image to upload queue, this will validate the image size and send POST request to upload image
+                imageView.$('.upload-button-input').fileupload('add', {files: [createFakeImageFile(60)]});
+
+                // Verify image upload progress message
+                expect(imageView.$('.upload-button-title').text().trim()).toBe(imageView.titleUploading);
+
+                $(window).trigger('beforeunload');
+
+                expect(imageView.onBeforeUnload).toHaveBeenCalled();
+            });
+
+            it('renders message correctly', function() {
```

Comment:
Nit: this test isn't about the learner_profile_fields. I'd move it into a message_banner_spec.js file.